pkg/apis/resource/validation/validation.go

```diff
@@ -738,16 +742,37 @@ func validateRawExtension(rawExtension runtime.RawExtension, fldPath *field.Path
 	return allErrs
 }
 
+const interfaceNameMaxLength int = 256
+const hardwareAddressMaxLength int = 128
+
 func validateNetworkDeviceData(networkDeviceData *resource.NetworkDeviceData, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if networkDeviceData == nil {
 		return allErrs
 	}
-	allErrs = append(allErrs, validateSlice(networkDeviceData.Addresses, -1,
+
+	if len(networkDeviceData.InterfaceName) > interfaceNameMaxLength {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("InterfaceName"), networkDeviceData.InterfaceName, validation.MaxLenError(interfaceNameMaxLength)))
+	}
+
+	if len(networkDeviceData.HardwareAddress) > hardwareAddressMaxLength {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("HardwareAddress"), networkDeviceData.HardwareAddress, validation.MaxLenError(hardwareAddressMaxLength)))
+	}
+
+	allErrs = append(allErrs, validateSet(networkDeviceData.Addresses, -1,
 		func(address string, fldPath *field.Path) field.ErrorList {
-			var allErrs field.ErrorList
-			allErrs = append(allErrs, validation.IsValidCIDR(fldPath, address)...)
-			return allErrs
-		}, fldPath.Child("addresses"))...)
+			return validation.IsValidCIDR(fldPath, address)
+		},
+		func(address string) (string, string) {
+			// reformat CIDR to handle different ways IPs can be written
+			// (e.g. 2001:db8::1/64 == 2001:0db8::1/64)
+			ip, ipNet, err := netutils.ParseCIDRSloppy(address)
+			if err != nil {
+				return "", "" // will fail at IsValidCIDR
+			}
+			maskSize, _ := ipNet.Mask.Size()
```

Comment:
once you have an ipNet, can't you just call `.String()` on that and get the normalized rendering?

Reply:
If you have `192.168.1.15/24`, ipNet.String() will return you `192.168.1.0/24`

Reply:
Ahh, righ, of course.  That's a trap that is SURE to bite us at some point.

Reply:
hmm, I was in favor of this format but after seen this now I have doubts ... seems the new netip library handles this correctly

https://go.dev/play/p/qCu9Rvs-dK9

Reply:
We just need to remember not to parse these into net.NetIP values.

Reply:
I prefer if we move to netip https://github.com/kubernetes/kubernetes/issues/106161 and , by doing that, we get rid of the forked IPSloppy parsers 🥇

Reply:
I haven't looked into the details of what you are doing here, but I remember that Jordan was very wary of relying on an external dependency for validating a Kubernetes API. That is why the DRA CEL library doesn't use the semver library to validate and instead uses a builtin regular expression.

Please ignore if it's not relevant...

Reply:
we are safe here in that regard, these are from std library ,  thanks for calling that out

Reply:
I still think is better to use `netip.ParsePrefix` , it returns the canonical form , does not have to use the old forked parsers and handles natively the CIDR format for IP addresses

Reply:
independently, this is not a blocker or anything,